### PR TITLE
[WIP] Feature/mutations

### PR DESCRIPTION
**Pull request by jimhommes**
Creates a new StackGraph, a VariationGraph, that defines all variations with an algorithm.

### CODE REVIEW

File: src/main/java/nl/tudelft/context/controller/DefaultNewickController.java [hunk 1]
```diff
@@ -1,6 +1,5 @@
 package nl.tudelft.context.controller;
 
-import javafx.beans.property.ObjectProperty;
```
Comment (Vennik) [inline src/main/java/nl/tudelft/context/controller/DefaultNewickController.java]: Checkstyle.

**Comment by Vennik** [comment]
What do you mark as variations and why?

**Comment by jimhommes** [comment]
Now works. The variations selected are all nodes that split and come together again. You have to look carefully at parts you think at first are wrong: It might be a combination of indents and larger variations. 

This current commit is debug so you can see how the debug works. After shipit I will remove the view and fix remaining checkstyle errors and whatnot. This should be the first step towards collapsing all variations and determining the mutations.

**Comment by Vennik** [comment]
2 println statements in variation graph.

**Comment by Vennik** [comment]
It's still just a marker without collapsing. If you make that possible I will gladly merge.

File: geex-core/src/main/java/nl/tudelft/context/drawable/graph/VariationLabel.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    /**
+     * Initialize the Label without the BaseLabels shown.
+     *
+     * @return Initialized Upper label
+     */
+    private Label initMainLabel() {
+
+        final Label label = new Label("" + node.getId());
```
Comment (Vennik) [inline geex-core/src/main/java/nl/tudelft/context/drawable/graph/VariationLabel.java:52]: Use Integer.toString.

File: geex-core/src/main/java/nl/tudelft/context/drawable/graph/VariationLabel.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import nl.tudelft.context.drawable.graph.DrawableGraphNodeLabel;
+import nl.tudelft.context.drawable.graph.DrawableNodeLabel;
+import nl.tudelft.context.model.graph.GraphNode;
+import nl.tudelft.context.model.graph.Node;
+
+/**
+ * @author Jim
+ * @since 6/4/2015
+ */
+public class VariationLabel extends DrawableNodeLabel {
```
Comment (Vennik) [inline geex-core/src/main/java/nl/tudelft/context/drawable/graph/VariationLabel.java:16]: Massive code duplication, just 1 style class difference with DrawableNodeLabel.

File: geex-core/src/main/java/nl/tudelft/context/drawable/graph/DrawableNode.java [hunk 4]
```diff
@@ -17,10 +18,16 @@
      */
     public DrawableNode(final Node node) {
         super(node);
+        isVariation = node.getVariation();
     }
 
     @Override
     public AbstractLabel getLabel(final MainController mainController, final AbstractGraphController graphController) {
-        return new DrawableNodeLabel(mainController, graphController, this, (Node) getNode());
+        if(isVariation && graphController.getGraphList().get(graphController.getDepth()).getMarkVariations()) {
```
Comment (Vennik) [inline geex-core/src/main/java/nl/tudelft/context/drawable/graph/DrawableNode.java:26]: Use the way graphnode and node labels are creates to do create different labels.

**Comment by jimhommes** [comment]
WIP

**Pull request closed without merging**